protocols/bftcosi/bftcosi.go

```diff
@@ -3,33 +3,38 @@
 package bftcosi
 
 import (
-	"math"
+	"crypto/sha512"
 	"sync"
 
-	"github.com/dedis/cothority/cosi"
+	"gopkg.in/dedis/cothority.v0/lib/dbg"
+
 	"github.com/dedis/cothority/log"
 	"github.com/dedis/cothority/sda"
 	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/cosi"
 )
 
 // VerificationFunction can be passes to each protocol node. It will be called
 // (in a go routine) during the (start/handle) challenge prepare phase of the
 // protocol. The passed message is the same as sent in the challenge phase.
-type VerificationFunction func(Msg []byte) bool
+type VerificationFunction func(Msg []byte, Data []byte) bool
 
 // ProtocolBFTCoSi is the main struct for running the protocol
 type ProtocolBFTCoSi struct {
 	// the node we are represented-in
 	*sda.TreeNodeInstance
+	// The message that will be signed by the BFTCosi
 	Msg []byte
+	// Data going along the msg to the verification
+	Data []byte
```

Comment:
Usually the services are supposed to being able to pass data along an instantiation of the protocol. But that part is not done yet, so the most easy one was to add this data-field here. This should go away as soon as the services can send messages for the instantiation of a protocol. See also #455